### DISCUSSION TITLE
Does Linear Programming solvers like GLOP allow ifelse condition?

### DISCUSSION THREAD

**Discussion by aihiangng** [google/or-tools#3453] [Linear Solver questions] [answered]
CP-SAT has the OnlyEnforceIf. Was wondering if LP supports such conditions? 
E.g if another variable is x, then y= 10, otherwise y=0.

**Comment by lperron** [comment, marked as answer]
It is impossible. It does not even enforce boolean variable to take 0 or 1 value.

To say differently, GLOP is pseudo-polynomial. adding only enforce if makes the problem NP-Hard.

**Reply by aihiangng** [reply]
@lperron thanks for replying! Is it impossible to model ifelse condition for LP in general or only for GLOP?

**Comment by lperron** [comment]
Lp in general.

**Reply by aihiangng** [reply]
@lperron If LP cannot ifelse condition, then why does Mixed Integer Linear Programming support ifelse condition? 
IP relies on LP relaxation and branch-and-cut to find optimal integer solutions. Im struggling to see why IP can but LP cannot even tho IP uses LP but with relaxation. 
thanks!

**Reply by Krabii** [reply]
An if-else condition is by nature a 0-1 decision, thus an integer one.
